sdks/java/io/google-cloud-platform/src/test/java/org/apache/beam/sdk/io/gcp/bigquery/BigQueryIOTest.java

```diff
@@ -394,6 +436,91 @@ public Job getJob(JobReference jobRef) throws InterruptedException {
     }
   }
 
+  /** A fake dataset service that can be serialized, for use in testReadFromTable. */
+  private static class FakeDatasetService implements DatasetService, Serializable {
+    private final com.google.common.collect.Table<String, String, Map<String, byte[]>> tables =
```

Comment:
Ditto on using readObject/writeObject so you don't need special logic inside the getters/setters